store/todos.js

```diff
@@ -41,6 +42,13 @@ export const mutations = {
   toggle(state, todo) {
     todo.done = !todo.done
   },
+  complete(state, todoId) {
+    const todoIndex = state.list.findIndex((t) => t.id === todoId)
+    const todo = state.list[todoIndex]
+
+    todo.completedAt = new Date()
+    state.list.splice(todoIndex, 1, todo)
+  },
```

Comment:
nitpicking: we could also use something like `map` to do this whole thing in one go!

Reply:
resolved